pkg/pool/operations/create.go

```diff
@@ -49,35 +49,37 @@ func (oc *OperationsConfig)Create(csp *cstor.CStorPoolInstance) error {
 	// 1. zpool create newpool mirror v0 v1
 	// 2. zpool add newpool log mirror v4 v5
 	// 3. zpool add newpool mirror v2 v3
-	spec := csp.Spec.DeepCopy()
-	raidGroups := spec.DataRaidGroups
-	for i, r := range raidGroups {
-			// we found the main raidgroup. let's create the pool
-			err = oc.createPool(csp, r)
-			if err != nil {
-				return errors.Errorf("Failed to create pool {%s} : %s",
-					PoolName(csp), err.Error())
-			}
-			// Remove this raidGroup
-			raidGroups = append(raidGroups[:i], raidGroups[i+1:]...)
-			break
+	cspiCopy := cspi.DeepCopy()
+	for i, r := range cspiCopy.Spec.DataRaidGroups {
+		// we found the main raidgroup. let's create the pool
+		err = oc.createPool(cspiCopy, r)
+		if err != nil {
+			return errors.Errorf("Failed to create pool {%s} : %s",
+				PoolName(), err.Error())
+		}
+		// Remove this raidGroup
+		cspiCopy.Spec.DataRaidGroups = append(cspiCopy.Spec.DataRaidGroups[:i], cspiCopy.Spec.DataRaidGroups[i+1:]...)
+		break
```

Comment:
Breaking without condition? Failing to understand the code. Existing code only but not able to understand.